docs/source/python/dataset.rst

```diff
@@ -15,6 +15,16 @@
 .. specific language governing permissions and limitations
 .. under the License.
 
+.. ipython:: python
+    :suppress:
+
+    # set custom tmp working directory for files that create data
+    import os
+    import tempfile
+
+    temp_working_dir = tempfile.mkdtemp(prefix="pyarrow-")
+    os.chdir(temp_working_dir)
```

Comment:
Will this clean it up at the end? Having multiple tempdirs accumulated when doing builds is not very nice.

Reply:
I currently didn't add an explicit clean-up (but could add a `shutil.rmtree(temp_working_directory)` at the end of each of the files using this), but so implicitly relied on the "temporary" nature / the OS to do this clean-up for us.

Personally I find that sufficient (building the docs seems to already add lots of "profile_..." dirs to /tmp, so having a few more "pyarrow-..." directories in there doesn't seem to matter much), but as mentioned also easy to add an explicit rmtree (it might still be good practice to clean-up even when using temporary files?).

Reply:
> building the docs seems to already add lots of "profile_..." dirs to /tmp, so having a few more "pyarrow-..." directories in there doesn't seem to matter much

Well, it would be nice if we could get rid of the "profile" dirs as well, do you know where they come from?

Reply:
> Well, it would be nice if we could get rid of the "profile" dirs as well, do you know where they come from?

I suppose from IPython, but would need to look into more detail